packages/next/next-server/lib/head.tsx

```diff
@@ -140,8 +140,11 @@ function reduceComponents(
     .filter(unique())
     .reverse()
     .map((c: React.ReactElement<any>, i: number) => {
-      const key = c.key || i
-      return React.cloneElement(c, { key })
+      let props: { [key: string]: any } = { key: c.key || i }
+      Object.entries(c.props).forEach(([key, val]) => {
+        if (![val, typeof val].includes('undefined')) props[key] = val
```

Comment:
I wrote this to also filter out `'undefined'`—maybe that's overreaching on the framework's part? Could also just be:
```suggestion
        if (val !== 'undefined') props[key] = val
```

Reply:
```suggestion
        if (typeof val !== 'undefined') props[key] = val
```